lib/puppet/face/config.rb

```diff
@@ -202,7 +208,13 @@ def report_section_and_environment(section_name, environment_name)
       if Puppet::FileSystem.exist?(path)
         Puppet::FileSystem.open(path, nil, 'r+:UTF-8') do |file|
           Puppet::Settings::IniFile.update(file) do |config|
-            setting_string = config.delete(options[:section], name)
+            if options[:section] == "master"
+              setting_string = config.delete(options[:section], name)
+              # also delete the setting from server
+              config.delete("server", name)
+            else
+              setting_string = config.delete(options[:section], name)
+            end
```

Comment:
If the user-specified section is `master` or `server`, then I think we want to delete from both `server` and `master` sections.